### ISSUE TITLE
RFE: switch to exit with non-zero status if warnings issued

### ISSUE THREAD

**Issue by smithfred** [rpm-software-management/rpmlint#122]
It would be great to have a switch (e.g. `--strict` or `--fail-on-warnings`) that caused any warnings to result in a non-zero exit code. I couldn't find any existing option for this.

Currently:
```sh
> rpmlint somefile
x packages and x specfiles checked; 0 errors, [>0] warnings.
> echo $?
0
```

With a new switch:
```sh
> rpmlint --strict somefile
x packages and x specfiles checked; 0 errors, [>0] warnings.
> echo $?
68
```

This would be useful for (interactive) scripting, to avoid having to parse the rpmlint summary (e.g. to implement a "linting warnings, continue?" feature in a script).

**Comment by Conan-Kudo** [comment]
Fixed for upcoming 2.0 release.

**Comment by brianjmurrell** [comment]
Does this `--strict` switch differentiate the exit code based on whether errors were found or just warnings?

Ideally I want to know, even with strict enabled, if errors were found or just warnings.  So I want 3 exit codes.  `0` for success, something `!0` for warnings and something else for errors.

**Comment by brianjmurrell** [comment]
Hrm.  Taking a look at the [commit](https://github.com/rpm-software-management/rpmlint/commit/6ffe00aa9715fe9e6a689b557b03ea71bca908ad), it seems not.  It seems it just upgrades all warnings to errors.  Pity.
